core/src/main/java/io/smallrye/openapi/runtime/io/ModelIO.java

```diff
@@ -330,7 +307,7 @@ protected Object readJson(V node, DataType desiredType) {
             } else if (desiredType.clazz == Object.class) {
                 return jsonIO().fromJson(node);
             } else {
-                // Log node dropped?
+                IoLogging.logger.invalidJsonType(desiredType.clazz.getName(), String.valueOf(node));
```

Comment:
`desiredType.clazz` could be `null` here

Reply:
This is true of both logging statements. If `desiredType.type` is `LIST` or `MAP`, then `desiredType.clazz` will be `null`.

Reply:
Fix for this is pushed - added a `DataType#toString` method.